src/psyclone/psyGen.py

```diff
@@ -988,8 +882,16 @@ def gen_code(self, parent):
         from psyclone.f2pygen import UseGen, DeclGen, AssignGen, CommentGen, \
             IfThenGen, CallGen
 
+        # The gen_code method generates new Symbol names, however, we want
+        # subsequent calls to invoke.gen_code() to produce the exact same code.
+        # To make this possible gen_code uses a duplicate of the symbol_table
+        # instead of the original one (so new symbols are not conserved outside
+        # gen_code). The original symbol table is restored at the end of this
```

Comment:
Please see my question about this which is associated with the previous review (but is still open).

Reply:
Replied the comment